graylog2-server/src/main/java/org/graylog2/rest/resources/roles/RolesResource.java

```diff
@@ -0,0 +1,261 @@
+import org.graylog2.users.RoleImpl;
+import org.graylog2.users.RoleService;
+import org.joda.time.DateTimeZone;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.inject.Inject;
+import javax.validation.Valid;
+import javax.validation.constraints.NotNull;
+import javax.ws.rs.BadRequestException;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.DELETE;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.PUT;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import java.net.URI;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+
+import static com.google.common.base.MoreObjects.firstNonNull;
+import static javax.ws.rs.core.Response.status;
+
+@RequiresAuthentication
+@Path("/roles")
+@Consumes(MediaType.APPLICATION_JSON)
+@Produces(MediaType.APPLICATION_JSON)
+@Api(value = "Roles", description = "User roles")
+public class RolesResource extends RestResource {
+    private static final Logger log = LoggerFactory.getLogger(RolesResource.class);
+
+    private final RoleService roleService;
+    private final InMemoryRolePermissionResolver inMemoryRolePermissionResolver;
+
+    @Inject
+    public RolesResource(RoleService roleService, InMemoryRolePermissionResolver inMemoryRolePermissionResolver) {
+        this.roleService = roleService;
+        this.inMemoryRolePermissionResolver = inMemoryRolePermissionResolver;
+    }
+
+    @GET
+    @RequiresPermissions(RestPermissions.ROLES_READ)
+    @ApiOperation(value = "List all roles", notes = "")
+    public RolesResponse listAll() throws NotFoundException {
+        final Set<Role> roles = roleService.loadAll();
+        Set<RoleResponse> roleResponses = Sets.newHashSet();
+        for (Role role : roles) {
+            roleResponses.add(RoleResponse.create(role.getName(), Optional.fromNullable(role.getDescription()), role.getPermissions(), role.isReadOnly()));
+        }
+
+        return RolesResponse.create(roleResponses);
+    }
+
+    @GET
+    @Path("{rolename}")
+    @ApiOperation("Retrieve permissions for a single role")
+    public RoleResponse read(@ApiParam(name = "rolename", required = true) @PathParam("rolename") String name) throws NotFoundException {
+        checkPermission(RestPermissions.ROLES_READ, name);
+
+        final Role role = roleService.load(name);
+        return RoleResponse.create(role.getName(), Optional.fromNullable(role.getDescription()), role.getPermissions(), role.isReadOnly());
+    }
+
+    @POST
+    @RequiresPermissions(RestPermissions.ROLES_CREATE)
+    @ApiOperation(value = "Create a new role", notes = "")
+    public Response create(@ApiParam(name = "JSON body", value = "The new role to create", required = true) @Valid @NotNull RoleResponse roleResponse) {
+        try {
+            roleService.load(roleResponse.name());
```

Comment:
yup, I think I just did it because the other services worked the same way, but I like `exists()` better